### PR TITLE
more linting

### PR DESCRIPTION
**Pull request by fperrad**

### CODE REVIEW

File: bn_mp_clear.c [hunk 1]
```diff
@@ -7,7 +7,7 @@
 void mp_clear(mp_int *a)
 {
    /* only do anything if a hasn't been freed previously */
-   if (a->dp != NULL) {
+   if ((a != NULL) && (a->dp != NULL)) {
```
Comment (minad) [inline bn_mp_clear.c]: a==NULL should not be handled here since it is a programming error.

File: bn_mp_reduce_is_2k.c [hunk 2]
```diff
@@ -29,8 +29,10 @@ mp_bool mp_reduce_is_2k(const mp_int *a)
             iz = 1;
          }
       }
+      return MP_YES;
+   } else {
+      return MP_YES;
```
Comment (minad) [inline bn_mp_reduce_is_2k.c]: Why?

File: bn_mp_clear.c (hunk 1 above)
Comment (fperrad) [reply bn_mp_clear.c]: in **libtomcrypt**, we could write :
```
LTC_ARGCHK(a != NULL);
```

but sadly **libtommath** have no equivalent.

Comment (minad) [reply bn_mp_clear.c]: Yes, we don't have such assertions. However the change you made masks a programming error and is therefore not good. You should revert this change.

Comment (minad) [reply bn_mp_clear.c]: If you want to harden the library, what I would recommend - define MP_ASSERT(x) in tommath.h which can be optionally activated in hardened/debugging mode. Then add MP_ASSERTS where it makes sense.

Comment (minad) [reply bn_mp_clear.c]: I just checked LTC, and they are doing exactly what I suggested - aborting in the case of assertion failure, but not masking the error and just returning without any message.

**Comment by fperrad** [comment]
commit on `bn_mp_clear.c` removed

**Pull request merged**